tls/s2n_client_hello.c

```diff
@@ -551,14 +552,17 @@ int s2n_process_client_hello(struct s2n_connection *conn)
     const struct s2n_ecc_preferences *ecc_pref = NULL;
     POSIX_GUARD(s2n_connection_get_ecc_preferences(conn, &ecc_pref));
     POSIX_ENSURE_REF(ecc_pref);
-    POSIX_ENSURE_GT(ecc_pref->count, 0);
-    if (s2n_ecc_preferences_includes_curve(ecc_pref, TLS_EC_CURVE_SECP_256_R1)) {
-        conn->kex_params.server_ecc_evp_params.negotiated_curve = &s2n_ecc_curve_secp256r1;
+
+    if (ecc_pref->count == 0) {
```

Comment:
What happens when a client indicates no preference and our security policy doesn't contain ECC preferences? Do we need to set a kem group fallback here? Or will that happen somehow anyway?